API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+using static System.Reflection.Metadata.BlobBuilder;
+
+namespace API.Controllers
+{
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = "IsAllUsers")]
+    [Route("api/vl")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookService _bookService;
+
+        public BooksController(IBookService bookService)
+        {
+            _bookService = bookService;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns>A list of all books</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> GetBooks()
+        {
+            var books = await _bookService.GetBooksAsync();
+            if (books.Count == 0) return NotFound();
+            return Ok(books);
+        }
+
+        /// <summary>
+        /// Get a specific book by id
+        /// </summary>
+        /// <param name="id">id</param>
+        /// <returns>Book</returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesDefaultResponseType]
+        public async Task<ActionResult<Book>> GetBook(int id)
```

Comment:
BookDto

Reply:
Why?

Reply:
https://learn.microsoft.com/en-us/aspnet/web-api/overview/data/using-web-api-with-entity-framework/part-5

To establish the pattern of using a DTO and avoid returning an Entity to the frontend for reasons listed in there

Reply:
Are you saying the parameter should be bookDto? I am already returning a bookDto?

Reply:
Not according to your controller's signature

Reply:
ActionResult<Book>